src/Symfony/Component/HttpKernel/Debug/TraceableEventDispatcher.php

```diff
@@ -472,11 +472,13 @@ private function unwrapListener($listener, $eventId)
         // get the original listener
         if (is_object($listener)) {
             if (null === $eventId) {
-                foreach (array_keys($this->wrappedListeners) as $eventId) {
-                    if (isset($this->wrappedListeners[$eventId][$listener])) {
-                        return $this->wrappedListeners[$eventId][$listener];
+                foreach ($this->wrappedListeners as $eventId => &$eventListeners) {
```

Comment:
Removed. Since $eventListeners is not a reference anymore, complementary unset is also removed.